### DISCUSSION TITLE
Abr files / settings  (Tutorial : how to save and auto load custom brush settings presets)

### DISCUSSION THREAD

**Discussion by calintz9999** [photopea/photopea#6376] [Ideas]
I've been scratching my head at how to explain this one.

1) It was super confusing to understand that I must first add a new abr file to then be able to add my own brushes in there.
Since I'm a PS user, I never really had to think about settings being remembered when I edit default brushes, it happened naturally and gets saved in the preference folders. But here with a browser, it's all different so I tried everything with different browsers (Edge, chrome without extensions, etc..) and I couldn't figure out how to keep my brushes with my settings .... this led me to part 2)

2) Once your refresh the page, your brushes or brush settings will be gone. 
Also and more importantly, nothing gets "stored" between default brushes, go and try yourself , all settings disappear, 
Click on any default brush, tick "tip dynamics, transfer with control = pen pressure" and click on another brush then come back to the previous brush = settings are all gone , nothing stored.
So it's either : 
-a  bug = meaning ticking "tip dynamics" is supposed to stay ticked but it doesnt, ( @photopea : I'll open an issue if needed)
-a browser limitation = for some reason , this is not supported by browsers and we need a fix to avoid the frustration 

*** For photopea Dev :  
- when we edit a default brush /  add a new brush it gets automatically added to a new abr file of our own to avoid any browser limitations
- when editing a brush : Warn user that nothing will be saved if tool is changed 
- when adding a new brush to default abr file : warn user that nothing will be saved when page is refreshed.
======================================================
WORKAROUND FIX FOR BRUSH SETTINGS SAVED AND AUTO LOADED /// 
Once I understood what was going on I was finally able to finally create save and automatically load my own brushes with my own settings saved and loaded. 

Here is how it goes : 
step 1 : go to preset manager and  select a brush , any brush  and export a temporary abr file
step 2 : Import that abr file and click brush icon and select that brush in your imported abr file ( temporary abr file)
step 3 : with that brush selected in your temporary abr file, set opacity on pen pressure, same for flow and size , etc
and create your own brush from the brush settings > Define brush
step 4 : now that new brush you just created gets added to your own abr file, repeat this process for any number of brushes
you need 
step 5 : go to brush settings and select all the brushes of your temporary abr file and export a final abr file 
step 6 : now delete all abr file already present from your preset manager and refresh the page 
step 7 : import your final abr file (the one containing all your custom created brushes) and of course say OK when prompted to auto load for further session 
step 8 : Now if you click brushes and go to your final abr file, select one of your own customized brush, the settings will be kept for all your customized brushes

The only caveat is : you have to remember which brush is for which tool, I basically created one for brush tool, one for eraser tool and one for smudge tool, you can rename the brush based on each tool but if you have many you might have to check the names of the brushes to be sure what's what...

step 9 : That's it now your own brushes with your own settings are preloaded but you still have to select 
a brush per tool,

*** @photopea :  It would also be cool to have the name of the brushes show, maybe have some kind of list feature  
so we can see the brush names . This way it would be easier to select them based on each tool once the page is refreshed.

3)*** @photopea  :  
 It would also be great to be able to rename abr folders as we can already rename brushes themselves.

**Comment by photopea** [comment]
When you open www.Photopea.com, it automatically loads the ABR with 14 default brushes. Let's say the user deletes four brushes and adds 10 new brushes. Then, the user closes Photopea and opens it again the next day. What should they see?

- See the 14 default brushes 
-- but their own brushes will be lost
- See the 20 brushes from the last day
-- but it will be impossible to get back the default brushes (which they deleted)
-- we will store ABR files on every computer which opens www.Photopea.com even for 1 second

Now, we have a Local Storage mechanism, which allows users to overcome the problems with the first solution.

**Reply by calintz9999** [reply]
I agree , having your own abr file is the way to go but then it should be the way new brushes are created automatically without having to do my work around two step solution. 
Meaning Photopea should automatically add your own brushes in a separate abr file when click on "Define new" .... instead of giving the opportunity to add them to the default abr file only to loose them once you refresh the page .
